test/extensions/filters/common/expr/expr_context_speed_test.cc

```diff
@@ -0,0 +1,171 @@
+#include "source/common/http/header_map_impl.h"
+#include "source/common/network/address_impl.h"
+#include "source/common/protobuf/protobuf.h"
+#include "source/common/router/string_accessor_impl.h"
+#include "source/extensions/filters/common/expr/context.h"
+
+#include "test/mocks/local_info/mocks.h"
+#include "test/mocks/ssl/mocks.h"
+#include "test/mocks/stream_info/mocks.h"
+#include "test/mocks/upstream/host.h"
+#include "test/test_common/utility.h"
+
+#include "benchmark/benchmark.h"
+#include "gmock/gmock.h"
+
+namespace Envoy {
+
+using testing::NiceMock;
+using testing::Return;
+
+class ExpressionContextSpeedTest {
+public:
+  ExpressionContextSpeedTest(size_t num_attributes) {
+    ssl_info_ = std::make_shared<NiceMock<Ssl::MockConnectionInfo>>();
+
+    // Set up request headers
+    request_headers_ = makeRequestHeaders();
+
+    // Set up response data
+    response_headers_ = makeResponseHeaders();
+    response_trailers_ = makeResponseTrailers();
+
+    // Set up stream info
+    setupStreamInfo();
+
+    // Set up filter state with specified number of attributes
+    setupFilterState(num_attributes);
+
+    // Create all wrappers
+    request_ = std::make_unique<Extensions::Filters::Common::Expr::RequestWrapper>(
+        arena_, &request_headers_, info_);
+    response_ = std::make_unique<Extensions::Filters::Common::Expr::ResponseWrapper>(
+        arena_, &response_headers_, &response_trailers_, info_);
+    filter_state_ = std::make_unique<Extensions::Filters::Common::Expr::FilterStateWrapper>(
+        arena_, *info_.filterState());
+  }
+
+  void testRequestAttributes(::benchmark::State& state) {
+    static const std::vector<absl::string_view> attributes = {
+        "scheme", "path", "method", "referer", "user-agent", "size", "total_size", "protocol"};
```

Comment:
you can use the string_views for these attributes here

Reply:
Done